tests/boot/boot_to_desktop.pm

```diff
@@ -32,7 +32,7 @@ sub run {
     # Do not attempt to log into the desktop of a system installed with SLES4SAP
     # being prepared for upgrade, as it does not have an unprivileged user to test
     # with other than the SAP Administrator
-    my $nologin = (get_var('HDDVERSION') and is_upgrade() and is_sles4sap());
+    my $nologin = (get_var('HDDVERSION') and is_upgrade() and is_sles4sap()) || get_var('HA_CLUSTER');
```

Comment:
Since you're using `||`, could you also change those `and`'s for `&&`.

Reply:
Done!